static/src/javascripts/projects/commercial/modules/dfp/enable-lazy-load.js

```diff
@@ -1,15 +1,27 @@
 define([
-    'common/utils/add-event-listener',
+    'common/utils/mediator',
     'commercial/modules/dfp/dfp-env',
     'commercial/modules/dfp/lazy-load'
-], function (addEventListener, dfpEnv, lazyLoad) {
+], function (mediator, dfpEnv, lazyLoad) {
+    /* observer: IntersectionObserver?. The observer used to detect when ad slots enter the viewport */
+    var observer = null;
+
     return enableLazyLoad;
 
-    function enableLazyLoad() {
+    function enableLazyLoad(advert) {
         if (!dfpEnv.lazyLoadEnabled) {
             dfpEnv.lazyLoadEnabled = true;
-            addEventListener(window, 'scroll', lazyLoad, { passive: true });
-            lazyLoad();
+            if (dfpEnv.lazyLoadObserve) {
+                observer = new IntersectionObserver(lazyLoad, { rootMargin: '200px 0%' });
+                dfpEnv.adverts.forEach(function (advert) {
+                    observer.observe(advert.node);
+                });
+            } else {
+                mediator.on('window:throttledScroll', lazyLoad);
+                lazyLoad();
+            }
+        } else if (dfpEnv.lazyLoadObserve && advert) {
+            observer.observe(advert.node);
```

Comment:
How so? If there is an incentive to measure that, then I can add a separate function, but otherwise I found this solution to have a minimal footprint. Or do you have a better solution in mind?